main/src/ca/uwaterloo/flix/language/ast/Scheme.scala

```diff
@@ -128,31 +127,29 @@ object Scheme {
     */
   // TODO can optimize?
   def equal(sc1: Scheme, sc2: Scheme, classEnv: Map[Symbol.ClassSym, List[Ast.Instance]])(implicit flix: Flix): Boolean = {
-    lessThanEqual(sc1, sc2, classEnv) && lessThanEqual(sc2, sc1, classEnv)
```

Comment:
If checkLessEqual returns a validation then these could be seqM'ed to combine the errors.